R/derive_param_confirmed_resp.R

```diff
@@ -0,0 +1,382 @@
+#' Derive Confirmed Response Parameter
+#'
+#' Derive confirmed response parameter
+#'
+#' @param dataset Input dataset
+#'
+#'   The `PARAMCD`, `ADT`, and `AVALC` variables and the variables specified by
+#'   `subject_keys` and `reference_date` are expected.
+#'
+#' @param dataset_adsl ADSL input dataset
+#'
+#'   The variables specified for `subject_keys` are expected. For each subject
+#'   of the specified dataset a new observation is added to the input dataset.
+#'
+#' @param filter_source Source filter
+#'
+#'   All observations in `dataset_source` fulfilling the specified condition are
+#'   considered for deriving the confirmed best overall response.
+#'
+#' @param source_pd Date of first progressive disease (PD)
+#'
+#'   If the parameter is specified, the observations of the input dataset for
+#'   deriving the new parameter are restricted to observations up to the
+#'   specified date. Observations at the specified date are included. For
+#'   subjects without first PD date all observations are take into account.
+#'
+#'   *Permitted Values:* a `date_source` object (see `admiral::date_source()`
+#'   for details)
+#'
+#'   *Default:* `NULL`,
+#'
+#' @param source_datasets Source dataset for the first PD date
+#'
+#'   A named list of datasets is expected. It links the `dataset_name` from
+#'   `source_pd` with an existing dataset.
+#'
+#'   For example if `source_pd = pd_date` with
+#'   ```{r eval=FALSE}
+#'   pd_date <- date_source(
+#'     dataset_name = "adrs",
+#'     date = ADT,
+#'     filter = PARAMCD == PD
+#'   )
+#'   ```
+#'   and the actual response dataset in the script is `myadrs`, `source_datasets
+#'   = list(adrs = myadrs)` should be specified.
+#'
+#'   This allows to define `pd_date` at a higher level, e.g., at company level,
+#'   where the actual dataset does not exist.
+#'
+#' @param ref_confirm Minimum time period for confirmation
+#'
+#'   The assessment and the confirmatory assessment for `"CR"` and `"PR"` have
+#'   to be at least the specified number of days apart.
+#'
+#' @param max_nr_ne
+#'
+#'   The specified number of `"NE"` assessments between the assessment and the
+#'   confirmatory assessment for `"CR"` and `"PR"` response is accepted.
+#'
+#'   *Permitted Values:* a non-negative numeric scalar
+#'
+#'   *Default:* `1`
+#'
+#' @param accept_sd Accept `"SD"` for `"PR"`?
+#'
+#'   If the argument is set to `TRUE`, one `"SD"` assessment between the
+#'   assessment and the confirmatory assessment for `"PR"` response is accepted.
+#'   Otherwise, no `"SD"` assessment must occur between the two assessments.
+#'
+#'   *Permitted Values:* a logical scalar
+#'
+#'   *Default:* `FALSE`
+#'
+#' @param aval_fun Function to map character analysis value (`AVALC`) to numeric
+#'   analysis value (`AVAL`)
```

Comment:
is this so people can choose whether they agree with N=0 and Y=1? i'm not sure if worth including. anoyone could always update that themselves downstream if so picky, but this is the kind of case i'd expect companies to have some pragmatism and just follow the admiral standard

Reply:
Yes, in this case it is very unlikely that something different than the default is used. I just added it to be consistent with the other functions.

BTW, looking back at the experience with implementing the Roche standard I am not so optimistic that the "admiral standard" is just followed. It seems that many people are willing to spend a considerable amount of time in implementing cosmetic deviations from the standard.

Reply:
Yes, in this case it is very unlikely that something different than the default is used. I just added it to be consistent with the other functions.

BTW, looking back at the experience with implementing the Roche standard I am not so optimistic that the "admiral standard" is just followed. It seems that many people are willing to spend a considerable amount of time in implementing cosmetic deviations from the standard.

Reply:
yes fair point! its ultimately a balance in the end as we need to try to ease the adoption by keeping as much as possible familiar to users, but then also striving for the greater good of industry harmonization whilst picking our battles with standards teams - and for which admiral alone will never achieve